From 03da6e73a7f2d732304715923dffc93a656689c7 Mon Sep 17 00:00:00 2001
From: Neena Busireddy <neenareddy.busireddy@nxp.com>
Date: Wed, 24 Sep 2025 13:12:33 -0500
Subject: [PATCH] apitrace: Add apitrace build issues related to GCC 15

https://gcc.gnu.org/gcc-15/porting_to.html#cxx
Some C++ Standard Library headers have been changed to
no longer include other headers that were being used
internally by the library.

Upstream-Status: Pending
Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>
---
 frametrim/ft_dependecyobject.hpp | 1 +
 retrace/metric_writer.hpp        | 1 +
 2 files changed, 2 insertions(+)

diff --git a/frametrim/ft_dependecyobject.hpp b/frametrim/ft_dependecyobject.hpp
index 2cd0b768..508b23cf 100644
--- a/frametrim/ft_dependecyobject.hpp
+++ b/frametrim/ft_dependecyobject.hpp
@@ -28,6 +28,7 @@
 #pragma once
 
 #include "ft_tracecall.hpp"
+#include <cstdint>
 
 namespace frametrim {
 
diff --git a/retrace/metric_writer.hpp b/retrace/metric_writer.hpp
index 9162878a..3cb0142e 100644
--- a/retrace/metric_writer.hpp
+++ b/retrace/metric_writer.hpp
@@ -28,6 +28,7 @@
 #include <queue>
 #include <string>
 #include <unordered_map>
+#include <cstdint>
 
 #include "metric_backend.hpp"
 #include "mmap_allocator.hpp"
-- 
2.34.1

